website/statistics.py

```diff
@@ -101,6 +104,103 @@ def get_class_stats(self, user, class_id):
+                                             matrix_values=matrix_values,
+                                             adventure_names=adventure_names,
+                                             student_adventures=student_adventures,
+                                             page_title=gettext("title_class grid_overview"),
+                                             )
+
+    @route("/grid_overview/class/<class_id>/level", methods=["POST"])
+    @requires_login
+    def change_checkbox(self, user, class_id):
+        level = request.args.get('level')
+        student_index = request.args.get('student_index', type=int)
+        adventure_index = request.args.get('adventure_index', type=int)
+        students, class_, class_adventures_formatted, ticked_adventures, adventure_names, _ = self.get_grid_info(
+            user, class_id, level)
+        matrix_values = self.get_matrix_values(students, class_adventures_formatted, ticked_adventures, level)
+
+        adventure_names = {value: key for key, value in adventure_names.items()}
+        current_adventure_name = class_adventures_formatted[level][adventure_index]
+        student_adventure_id = f"{students[student_index]}-{adventure_names[current_adventure_name]}-{level}"
+
+        self.db.update_student_adventure(student_adventure_id, matrix_values[student_index][adventure_index])
+        _, _, _, ticked_adventures, _, student_adventures = self.get_grid_info(user, class_id, level)
+        matrix_values[student_index][adventure_index] = not matrix_values[student_index][adventure_index]
+
+        return jinja_partials.render_partial("customize-grid/grid-levels.html",
+                                             level=level,
+                                             class_info={"id": class_id, "students": students, "name": class_["name"]},
+                                             current_page="grid_overview",
+                                             max_level=hedy.HEDY_MAX_LEVEL,
+                                             class_adventures=class_adventures_formatted,
+                                             ticked_adventures=ticked_adventures,
+                                             adventure_names=adventure_names,
+                                             student_adventures=student_adventures,
+                                             matrix_values=matrix_values,
+                                             page_title=gettext("title_class grid_overview"),
+                                             )
+
+    def get_matrix_values(self, students, class_adventures_formatted, ticked_adventures, level):
+        rendered_adventures = class_adventures_formatted.get(level)
+        grid_values = []
+        for student in students:
+            clone = [None for _ in range(len(class_adventures_formatted[level]))]
+            student_list = ticked_adventures.get(student)
+            if student_list and rendered_adventures:
```

Comment:
Im not sure if we can reduce the nesting here a bit

Reply:
Changed it a tiny bit. Let me know your thoughts on the new commit